src/ctapipe/monitoring/calculator.py

```diff
@@ -0,0 +1,336 @@
+        allow_none=True,
+        help=(
+            "List of dicts containing the name of the OutlierDetector subclass to be used, "
+            "the aggregated statistic value to which the detector should be applied, "
+            "and the validity range of the detector. "
+            "E.g. ``[{'apply_to': 'std', 'name': 'RangeOutlierDetector', 'validity_range': [2.0, 8.0]},]``."
+        ),
+    ).tag(config=True)
+
+    chunk_shift = Int(
+        default_value=None,
+        allow_none=True,
+        help=(
+            "Number of samples to shift the aggregation chunk for the calculation "
+            "of the statistical values. Only used in the second_pass(), since the "
+            "first_pass() is conducted with non-overlapping chunks (chunk_shift=None)."
+        ),
+    ).tag(config=True)
+
+    faulty_pixels_fraction = Float(
+        default_value=0.1,
+        allow_none=True,
+        help="Minimum fraction of faulty camera pixels to identify regions of trouble.",
+    ).tag(config=True)
+
+    def __init__(
+        self,
+        subarray,
+        config=None,
+        parent=None,
+        **kwargs,
+    ):
+        """
+        Parameters
+        ----------
+        subarray: ctapipe.instrument.SubarrayDescription
+            Description of the subarray. Provides information about the
+            camera which are useful in calibration. Also required for
+            configuring the TelescopeParameter traitlets.
+        config: traitlets.loader.Config
+            Configuration specified by config file or cmdline arguments.
+            Used to set traitlet values.
+            This is mutually exclusive with passing a ``parent``.
+        parent: ctapipe.core.Component or ctapipe.core.Tool
+            Parent of this component in the configuration hierarchy,
+            this is mutually exclusive with passing ``config``
+        """
+        super().__init__(subarray=subarray, config=config, parent=parent, **kwargs)
+        self.subarray = subarray
+
+        # Initialize the instances of StatisticsAggregator
+        self.stats_aggregators = {}
+        for _, _, name in self.stats_aggregator_type:
+            self.stats_aggregators[name] = StatisticsAggregator.from_name(
+                name, subarray=self.subarray, parent=self
+            )
+
+        # Initialize the instances of OutlierDetector
+        self.outlier_detectors = {}
+        if self.outlier_detector_list is not None:
+            for d, outlier_detector in enumerate(self.outlier_detector_list):
+                # Check if all required keys are present
+                missing_keys = {
+                    "apply_to",
+                    "name",
+                    "validity_range",
+                } - outlier_detector.keys()
+                if missing_keys:
+                    raise TraitError(
+                        f"Entry '{d}' in the ``outlier_detector_list`` trait is missing required key(s): {', '.join(missing_keys)}"
```

Comment:
This line looks too long... how did it escape the pre-commit checks?

Reply:
this is an interesting question... And we've seen some weird skips of pre-commit hook in Christoph's PR too...

Reply:
I don't know. To double-check I run 
```bash
$ pre-commit run --files $(git diff --name-only origin/main)
trim trailing whitespace.................................................Passed
check for added large files..............................................Passed
check for case conflicts.................................................Passed
check for merge conflicts................................................Passed
fix end of files.........................................................Passed
codespell................................................................Passed
ruff.....................................................................Passed
ruff-format..............................................................Passed
```
which `Passed`.

Reply:
I created an obvious style limitation in the file and it got picked up by the linter... so I am lost why this long line is "ok"...